### PR TITLE
chore: [M3-7551] - Update Vite and Vitest

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Updates Vite v4 to v5
  - Nothing super important to call out
- Updates Vitest to v1
  - Hoping to see performance improvements  

### Prerequisites
- Check out this PR
- Run `yarn` in the repo's root to ensure dependencies are up to date

### Verification steps 
- Verify the local dev server works (`yarn dev`)
- Verify all CI / E2Es pass
- Verify new coverage feature still works
- Check the preview link for issues

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "The Linode Manager website",
   "version": "1.107.0",
   "private": true,
+  "type": "module",
```
Comment (abailly-akamai) [inline packages/manager/package.json:7]: I _think_ this will be good for getting rid of legacy `.mjs` files?

**Review by abailly-akamai — APPROVED** [review]
All looked good when testing locally:
- `yarn` ✅ 
- `yarn build` ✅ 
- `yarn up` ✅ 
- `yarn test` ✅
- `yarn coverage` ✅
- `yarn coverage:summary` ✅
- Tried some e2e suites - no regression noticed  ✅

I noticed a ~10% dip in coverage as a result of this update. Not worried about it but something to keep in mind. Not sure what led this discrepancy. Perhaps they updated the parsing somehow.

**Comment by jdamore-linode** [comment]
I'm going to circle back with a proper review later, but just want to point out that the most recent E2E test run only has one failure and it can be disregarded (it's just the `resize-linode.spec.ts` warm resize failure that's just caused by a notification text change made recently on the API side) -- It think we can treat it as a pass.

**Review by jdamore-linode — APPROVED** [review]
Nice! Thanks @bnussman-akamai

**Comment by bnussman-akamai** [comment]
For those of us who use watch mode: https://twitter.com/vitest_dev/status/1733473742607511836

**Pull request merged**